implementation/src/main/java/io/smallrye/mutiny/helpers/spies/Spy.java

```diff
@@ -0,0 +1,40 @@
+package io.smallrye.mutiny.helpers.spies;
+
+import java.util.function.Predicate;
+
+import io.smallrye.mutiny.Uni;
+
+public interface Spy {
```

Comment:
Should we have UniSpy and MultiSpy to avoid name clash?

Reply:
Or maybe put everything in the same class as Uni / Multi would provide different signatures.

Reply:
Yes that's my idea (I initially drafted `UniSpy` then renamed to `Spy`). This way you can write `Spy.onEvent(uni)` which reads better.